packages/components/src/components/drawer/drawer.jsx

```diff
@@ -15,54 +15,45 @@ const IconDrawer = ({ className }) => (
     </svg>
 );
 
-class Drawer extends React.PureComponent {
-    constructor(props) {
-        super(props);
-        this.state = { is_open: props.is_open };
-    }
+const Drawer = ({
+    anchor = 'left',
+    children,
+    className,
+    is_open,
+    toggleDrawer,
+    contentClassName,
+    footer,
+    header,
+    width = 250,
+    zIndex = 4,
+}) => {
+    const [will_open, set_is_open] = React.useState(is_open);
 
-    componentWillReceiveProps(nextProps) {
-        this.setState({
-            is_open: nextProps.is_open,
-        });
-    }
+    React.useEffect(() => {
+        set_is_open(is_open);
+    }, [is_open]);
```

Comment:
Won't this cause an infinite loop?

Reply:
@yashim-deriv I passed `is_open` in the end  in `useEffect ` and if value change it should rerender because we use this component in different places.

Reply:
@yashim-deriv but I am not sure that use `useEffect` a good idea because here I use the same value `is_open `, what do you think?